aten/src/ATen/native/LinearAlgebra.cpp

```diff
@@ -685,6 +685,513 @@ Tensor& matmul_out(Tensor &result, const Tensor & tensor1, const Tensor & tensor
   return result;
 }
 
+// helper methods for matrix_exp
+namespace {
+
+template <typename scalar_t, int ROW, int COL>
+using array2d = std::array<std::array<scalar_t, COL>, ROW>;
+
+// we consider 6 Taylor expansions of degree
+// 1, 2, 4, 8, 12, 18
+constexpr int total_n_degs = 6;
+
+Tensor operator_1_norm(const Tensor& tensor) {
+  return std::get<0>(tensor.abs().sum(-2).max(-1));
+}
+
+// Allocates a buffers of uninitialized or zero values
+// of shape [n_copies, a.size()]
+Tensor _allocate_buffer(const Tensor& a, int n_copies, bool is_zero = false) {
+  if (is_zero) {
```

Comment:
nit: You could unconditionally allocate an `at::empty` tensor and then `zero_` it after if `is_zero`; it will be just as fast!